doc/specs/#11000 - Marks/Shell-Integration-Marks.md

```diff
@@ -0,0 +1,474 @@
+* [x] `clearMark`: Remove any marks in the selected region (or at the cursor
+  position)  (in [#12948])
+* [x] `clearAllMarks`: Remove all the marks from the buffer.  (in [#12948])
+
+
+#### Selecting commands & output
+
+_Inspired by a long weekend of manually copying .csv output from the Terminal to
+a spreadsheet, only to discover that we rejected [#4588] some years ago._
+
+* [x] `selectCommand(direction=[prev, next])`: Starting at the active selection
+  anchor, (or the cursor if there's no selection), select the command that
+  starts before/after this point (exclusive).  Probably shouldn't wrap around
+  the buffer.
+  * Since this will create a selection including the start of the command,
+    performing this again will select the _next_ command (in whatever
+    direction).
+* [x] `selectOutput(direction=[prev, next])`: same as above, but with command outputs.
+
+A convenient workflow might be a `multipleActions([selectOutput(prev),
+copy()])`, to quickly select the previous commands output.
+
+### Per-profile settings
+
+* [x] `autoMarkPrompts`: `bool`, default `false`.  (in [#12948])
+* [ ] `showFindMatchesOnScrollbar`: `bool`, default `true`.
+* [ ] `showMarksOnScrollbar`: `bool` or `flags({categories}...)` 
+  * As an example: `"showMarksOnScrollbar": ["error", "success"]`).
+  * Controls if marks should be displayed on the scrollbar. 
+  * If `true`/`"all"`, then all marks are displayed. 
+  * If `false`/`"none"`, then no marks are displayed. 
+  * If a set of categories are provided, only display marks from those categories.
+  * [x] the bool version is (in [#12948])
+  * [ ] The `flags({categories}...)` version is not done yet. 
+* [ ] `showGutterIcons`, for displaying gutter icons. 
+
+## UX Design
+
+An example of what colored marks look like:
+
+![Select the entire output of a command](https://user-images.githubusercontent.com/18356694/207696859-a227abe2-ccd4-4b81-8a2c-8a22219cd0dd.gif)
+
+This gif demos both prompt marks and marks for search results:
+
+![](https://user-images.githubusercontent.com/18356694/191330278-3f6bc207-1bd5-4ebd-bb0e-1f84b0170f49.gif)
+
+
+### Gutter icons
+
+![](vscode-shell-integration-gutter-mark.png)
+_An example of what the icons in the VsCode gutter look like_
+
+### Multiple marks on the same line
+
+When it comes to displaying marks on the scrollbar, or in the margins, the
+relative priority of these marks matters. Marks are given the following
+priority, with errors being the highest priority.
+* Error
+* Warning
+* Success
+* Prompt
+* Info (default)
+
+## Work needed to get marks to v1
+
+* [ ] Clearing the screen leaves marks behind
+  * [ ] Make sure `ED2` works to clear/move marks
+  * [ ] Same with `ED3`
+  * [ ] Same with `cls` / `Clear-Host`
+  * [ ] Clear Buffer action too.
+* [X] Circling doesn't update scrollbar
+  * I think this was fixed in [#14341], or in [#14045]
+* [ ] Resizing / reflowing marks
+* [ ] marks should be stored in the `TextBuffer`

```

Comment:
Will we come back and update the spec to check these boxes? Or should we remove the checks, track these in an issue, and call this the "Required for v1" list

Reply:
I should! I was hoping to have this merged so I could `x` these in future PRs